scripts/generate-config-data.js

```diff
@@ -0,0 +1,245 @@
+const https = require('https');
+const fs = require('fs');
+const path = require('path');
+
+function fetchSource () {
+  return new Promise(resolve => {
+    https.get('https://raw.githubusercontent.com/paritytech/parity/master/parity/cli/mod.rs', res => {
+      res.setEncoding('utf8');
+      let body = '';
+      res.on('data', data => { body += data; });
+      res.on('end', () => { resolve(body); });
+    });
+  });
+}
+
+function getCliOptions (source) {
+  const regex = /^\s*(FLAG|ARG) (\w+): *\([\w<>]+?\) *= (.+?)(?:u\d+|usize)?, or (.+),\n.+\n\s*"(.+)",(?:\s+\/\/.+)?$/gm;
+
+  const cliOptions = [];
+  for (const [, type, variableName, defaultValueString, configCallbackString, helpString] of execAll(regex, source)) {
+    cliOptions.push({type, variableName, defaultValueString, configCallbackString, helpString});
+  }
+
+  const parsedCliOptions = cliOptions.map(({defaultValueString, configCallbackString, helpString, ...rest}) => {
+    const defaultValue = parseDefaultValue(defaultValueString);
+    const help = helpString.replace(/\\"/g, '"');
+    const {configSection, configProp, configIs} = parseConfigCallback(configCallbackString);
+    return {defaultValue, help, configSection, configProp, configIs, ...rest};
+  });
+
+  return parsedCliOptions;
+}
+
+function* execAll (regex, str) {
+  const localCopy = new RegExp(regex, regex.flags);
+  let match;
+  while ((match = localCopy.exec(str))) {
+    yield match;
+  }
+}
+
+function parseDefaultValue (defaultValue) {
+  switch (defaultValue) {
+    case 'None':
+      return null;
+    case 'Vec::new()':
+      return [];
+    default:
+      try {
+        return JSON.parse(defaultValue);
+      } catch (e) {
+        console.warn('Warning: Failed to parse default value', defaultValue);
```

Comment:
Should it return something different than `undefined`?

Reply:
I think it's fine; also lets us distinguish between `null` default values and unrecognized (undefined) return values (although we could admittedly remove the `defaultValue` key altogether in the latter case).

The idea anyway is to provide a default value in data.extra.json for the config items that throw this warning.
I guess we could issue an error if a config item with unrecognized default value doesn't have a default value set in data.extra.json. I'll integrate this to the next commit.